lib/SILOptimizer/Mandatory/TFDevicePartition.cpp

```diff
@@ -151,6 +151,27 @@ void GraphFunctionDeviceInfo::handleDevicePlacement(
        SymbolicValue::getString(deviceString, ctx.getAllocator())});
 }
 
+DeviceType GraphFunctionDeviceInfo::chooseDevice(llvm::StringRef opType) const {
+  if (opType == "tfc.RecvFromHost" || opType == "tfc.SendToHost")
```

Comment:
Might be less verbose with llvm::StringSwitch?

Reply:
This is a short-term hack in any case. 

A better fix would be to use the code `opHasSideEffects |= valueKind == TFValueKind::ResourceHandle` in TFGraphFunctionLowering::visitGraphOperationInst(), or even better, use op device availability info once it's landed.